Sources/Sentry/SentryHub.m

```diff
@@ -254,7 +254,11 @@ - (SentryId *)captureEvent:(SentryEvent *)event withScope:(SentryScope *)scope
 
     transactionContext.sampled = [_sampler sample:samplingContext];
 
-    return [[SentryTracer alloc] initWithTransactionContext:transactionContext hub:self];
+    id<SentrySpan> tracer = [[SentryTracer alloc] initWithTransactionContext:transactionContext
+                                                                         hub:self];
+    [_scope setTransaction:tracer];
```

Comment:
H: we should offer a way to opt-out of that, otherwise every transaction will overwrite each other, see https://github.com/getsentry/sentry-java/pull/1313/files

Reply:
So true. But the default should be opt-in?
I like opt-out as default more.

Reply:
on Java, it bounds to the scope automatically unless you pass `false` to the flag, so opt out by default

Reply:
@brustolin but you should do an atomic reference swap (or lock access to the field if that's the only way to do it) essentially only if the field is `nil`. AS @marandaneto pointed out you don't want to override a reference set there before hand, potentially explicitly by the user while an auto instrumentation kicked-in. Logging would come in handy to point out whether we couldn't set something that was supposed to be set.

When removing it from there, do it only if the reference is still the same you're trying to close. 

For example: https://github.com/getsentry/sentry-dotnet/pull/880/files#diff-5ad21d586eb98c0917050a560662833d07ec0203d267f57efef308774d74311fR399-R400

```
        internal void ResetTransaction(ITransaction? expectedCurrentTransaction) =>
            Interlocked.CompareExchange(ref _transaction, null, expectedCurrentTransaction);
```

Reply:
Obj-c atomic properties already have atomic swap implementation.